soh/soh/Enhancements/randomizer/randomizer_grotto.c

```diff
@@ -128,8 +128,49 @@ static void Grotto_SetupReturnInfo(GrottoReturnInfo grotto, RespawnMode respawnM
     }
 }
 
+// Get the next entrance value while handling conversion of grotto rando IDs to real entrance values.
+// This method doesn't change player respawn data, so only use this if you are querying an entrance index.
+s16 Grotto_GetEntranceValueHandlingGrottoRando(s16 nextEntranceIndex) {
+    // Don't change anything unless grotto shuffle has been enabled
+    if (!Randomizer_GetSettingValue(RSK_SHUFFLE_GROTTO_ENTRANCES) && !Randomizer_GetSettingValue(RSK_SHUFFLE_OVERWORLD_SPAWNS) && !Randomizer_GetSettingValue(RSK_SHUFFLE_WARP_SONGS)) {
```

Comment:
Why are `RSK_SHUFFLE_OVERWORLD_SPAWNS` & `RSK_SHUFFLE_WARP_SONGS` checked in here?
If this is function is also used for other stuff it should probably be in `randomizer_entrance.c`.

Reply:
In it's simplest form, this method is just a copy of `Grotto_OverrideSpecialEntrance` but with all the state changing logic removed. So these checks existed the same over there. The original code is roughly 2 years old.

Spawns and warp songs are checked here because with grotto rando off, spawns or warp songs can still use a grotto location as an override, so they would be assigned the custom grotto entrance values that need to be handled by these functions.

As to why the original code was there rather in `randomizer_entrance.c` was because I wanted `Grotto_OverrideSpecialEntrance` to be "safe" to call on it's own without having to check those values externally. `Grotto_OverrideSpecialEntrance` is used from within the debug warp menu to handle grottos without running extra logic that resides in `Entrance_OverrideNextIndex`.

I'd be willing to refactor a lot of the entrance rando handling, but not in the scope of this fix.